lnrpc/routerrpc/router_server.go

```diff
@@ -427,10 +432,17 @@ func (s *Server) SendToRouteV2(ctx context.Context,
 	}
 
 	// Transform user errors to grpc code.
-	if err == channeldb.ErrPaymentInFlight ||
-		err == channeldb.ErrAlreadyPaid {
+	switch {
+	case errors.Is(err, channeldb.ErrPaymentExits):
```

Comment:
That this commit exists in isolation, does that mean that the previous commit wouldn't result in a working lnd?

Reply:
the itest can be run, so i guess it's working.

Reply:
Yes, but if it was already correct, this commit wouldn't be necessary by definition. To me it is an indication that it should really be part of something else that requires this for correctness (correctness in general, not just passing itests).

Reply:
> Yes, but if it was already correct, this commit wouldn't be necessary by definition.

Just to be clear, are you talking about the rpc changes, or just the changes made to `router_server.go`?